cluster/backup.go

```diff
@@ -0,0 +1,39 @@
+package cluster
+
+import (
+	"context"
+)
+
+func (c *Cluster) ETCDSave(ctx context.Context, snapshotName string) error {
+	if err := c.restore(); err != nil {
```

Comment:
lines 8-19 sequence gets repeated in several places, think it would make sense to extract it?